spark-jobs/src/main/scala/filodb/downsampler/BatchDownsampler.scala

```diff
@@ -362,5 +366,13 @@ object BatchDownsampler extends StrictLogging with Instance {
       blacklist.forall(w => !w.forall(pkPairs.contains))
     }
   }
+  /**
+    * This method is used for updating Partkey's schema to downsampled schema as gauge has different
+    * downsampled schema.
+    * schemaId is stored at offset+4.
+    */
+  private[downsampler] def updateDataSchema(partKeyBase: Array[Byte], partKeyOffset: Long, schemaId: Short): Unit = {
+    UnsafeUtils.setShort(partKeyBase, partKeyOffset + 4, schemaId)
```

Comment:
We need better abstraction here than this. This code will be affected if encoding of part keys change. Lets use a utility API that is closer to the part key encoding class.

Reply:
moved this method to RecordSchema